src/Open3D/Visualization/Visualizer/GuiVisualizer.cpp

```diff
@@ -236,6 +286,127 @@ GuiVisualizer::GuiVisualizer(
+        dlg->SetOnCancel([this]() { this->CloseDialog(); });
+        dlg->SetOnDone([this, renderScene](const char *path) {
+            this->CloseDialog();
+            auto newSky = GetRenderer().AddSkybox(ResourceLoadRequest(path));
+            if (newSky) {
+                impl_->lightSettings.hSky = newSky;
+
+                renderScene->SetSkybox(newSky);
+            }
+        });
+        ShowDialog(dlg);
+    });
+    loadButtons->AddChild(lightSettings.wgtLoadSky);
+    lightSettings.wgtBase->AddChild(loadButtons);
+
+    lightSettings.wgtBase->AddChild(gui::Horiz::MakeFixed(8));
+    lightSettings.wgtBase->AddChild(
+            std::make_shared<gui::Label>("Light switches:"));
+    auto checkboxes = std::make_shared<gui::Horiz>(0, gui::Margins{0, 0, 0, 8});
+    lightSettings.wgtAmbientEnabled =
+            std::make_shared<gui::Checkbox>("Ambient");
+    lightSettings.wgtAmbientEnabled->SetChecked(true);
+    lightSettings.wgtAmbientEnabled->SetOnChecked(
+            [this, renderScene](bool checked) {
+                if (checked) {
+                    renderScene->SetIndirectLight(impl_->lightSettings.hIbl);
+                } else {
+                    renderScene->SetIndirectLight(IndirectLightHandle());
+                }
+            });
+    checkboxes->AddChild(lightSettings.wgtAmbientEnabled);
+    lightSettings.wgtSkyEnabled = std::make_shared<gui::Checkbox>("Sky");
+    lightSettings.wgtSkyEnabled->SetChecked(true);
+    lightSettings.wgtSkyEnabled->SetOnChecked(
+            [this, renderScene](bool checked) {
+                if (checked) {
+                    renderScene->SetSkybox(impl_->lightSettings.hSky);
+                } else {
+                    renderScene->SetSkybox(SkyboxHandle());
+                }
+            });
+    checkboxes->AddChild(lightSettings.wgtSkyEnabled);
+    lightSettings.wgtDirectionalEnabled =
+            std::make_shared<gui::Checkbox>("Sun");
+    lightSettings.wgtDirectionalEnabled->SetChecked(true);
+    lightSettings.wgtDirectionalEnabled->SetOnChecked(
+            [this, renderScene](bool checked) {
+                renderScene->SetEntityEnabled(
+                        impl_->lightSettings.hDirectionalLight, checked);
+            });
+    checkboxes->AddChild(lightSettings.wgtDirectionalEnabled);
+    lightSettings.wgtBase->AddChild(checkboxes);
+
+    lightSettings.wgtIntensity = AddAttributeSlider(
+            gui::Slider::INT, 0.0, 1000000.0, lightDescription.intensity,
+            "Directional light intensity", lightSettings.wgtBase.get());
+    lightSettings.wgtIntensity->OnValueChanged =
+            [this, renderScene](double newValue) {
+                renderScene->SetLightIntensity(
+                        impl_->lightSettings.hDirectionalLight, newValue);
+            };
+    lightSettings.wgtAmbientIntensity = AddAttributeSlider(
+            gui::Slider::INT, 0.0, 1000000.0, kAmbientIntensity,
```

Comment:
Ambient intensity seems like it should be even lower, since it tends to come from multiple sides. Maybe 50000? Or 100000 at most?

Reply:
After tests it looks like 100000 is enough